blockly/src/server/Server.java

```diff
@@ -108,38 +118,86 @@ public class Server {
     "WandUnten",
     "naheWand"
   };
-  private static final Stack<String> currently_repeating_scope = new Stack<>();
+  private final Stack<String> currently_repeating_scope = new Stack<>();
 
   /**
    * Object containing the variable HUD in the dungeon. This object is used to add new variables to
    * the HUD. It will add int variables and arrays to the HUD and update existing varaible values.
    */
-  public static VariableHUD variableHUD = null;
+  public VariableHUD variableHUD = null;
 
   /**
    * Constructor of the server. Sets the hero.
    *
    * @param hero The hero entity. Used to control the movement of the hero.
    */
-  public Server(Entity hero) {
-    Server.hero = hero;
+  private Server(Entity hero) {
```

Comment:
Wird dieser Ctor irgendwo benutzt? (Zumindest in der GH-App sehe ich grad keine Verwendung?)

Aber es sollte zumindest einen privaten Ctor geben, damit niemand diese Klasse instantiieren kann. Damit es keine Verwirrungen gibt, würde ich lieber den parameterlosen Ctor dafür nehmen (außer dieser hier wird tatsächlich irgendwo aufgerufen - dann reicht das so).

Reply:
Der Server ist komplett abhängig von dem Hero. Aber alternative könnte man auf Game.hero() wechseln

Reply:
> Der Server ist komplett abhängig von dem Hero. 

ja, das schon.

aber der ctor kann nur noch von intern aufgerufen werden, und hier in der app sehe ich keinen aufruf... deshalb die frage. (kann aber gut sein, dass das ding irgendwo genutzt wird - müsste wohl doch noch den rechner anmachen.)

prinzipiell finde ich die übergabe der entity besser als es hart zu verdrahten. ist letztendlich wieder die diskussion etwas einfacherer code vs testbarkeit.

Reply:
> ... müsste wohl doch noch den rechner anmachen.

Ich wusste gar nicht das du den Rechner auch mal aus hast 😮😅

Also der Ctor wird nur hier verwendet:
```java
public static Server instance(Entity hero) {
  if (instance == null) {
    instance = new Server(hero);
  }
  return instance;
}
```